core/src/org/pentaho/platform/engine/core/system/PentahoSystem.java

```diff
@@ -867,98 +825,79 @@ public static Object putInGlobalAttributesMap( final Object key, final Object va
   public static Object removeFromGlobalAttributesMap( final Object key ) {
     return PentahoSystem.globalAttributes.remove( key );
   }
+  
+  @SuppressWarnings( "unchecked" )
+  private static void moveParametersFromRuntimeContextToSession( IRuntimeContext context, IPentahoSession session ) {
+    for ( String attributeName : (Set<String>) context.getOutputNames() ) {
+      IActionParameter output = context.getOutputParameter( attributeName );
 
+      Object data = output.getValue();
+      if ( data != null ) {
+        session.removeAttribute( attributeName );
+        session.setAttribute( attributeName, data );
+      }
+    }
+  }
+  
+  private static Map<String, IParameterProvider> createParameterProviderMap( final IPentahoSession session, IParameterProvider sessionParameters  ){
+    if ( sessionParameters == null ) {
+      sessionParameters = new PentahoSessionParameterProvider( session );
+    }
+    Map<String, IParameterProvider> parameterProviderMap = new HashMap<String, IParameterProvider>();
+    parameterProviderMap.put( SCOPE_SESSION, sessionParameters );
+    return parameterProviderMap;
+  }
+  
   public static void sessionStartup( final IPentahoSession session, IParameterProvider sessionParameters ) {
-
     List<ISessionStartupAction> sessionStartupActions =
         PentahoSystem.getSessionStartupActionsForType( session.getClass().getName() );
-    if ( sessionStartupActions == null ) {
-      // nothing to do...
+    if ( CollectionUtils.isEmpty( sessionStartupActions ) || !session.isAuthenticated() ) {
```

Comment:
Why are these changes being made for something that's supposed to be increasing test coverage ? This level of refactor needs to be done on a JIRA case.